src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingSearchHelpers.cs

```diff
@@ -0,0 +1,174 @@
+                typeArity = 0;
+            }
+            else
+            {
+                typeNameWithoutArity = typeName[0..backtickIndex];
+                typeArity = int.Parse(typeName[(backtickIndex + 1)..], CultureInfo.InvariantCulture);
+            }
+        }
+
+        private static async Task<ImmutableArray<UnitTestingDocumentSpan>> GetSourceLocationsInProcessAsync(
+            Project project,
+            UnitTestingSearchQuery query,
+            CancellationToken cancellationToken)
+        {
+            var (container, symbolName, symbolArity) = BuildQueryInfo(query);
+
+            var tasks = project.Documents.Select(d => GetSourceLocationsInProcessAsync(d, container, symbolName, symbolArity, query, cancellationToken));
+            var result = await Task.WhenAll(tasks).ConfigureAwait(false);
+            return result.SelectMany(r => r).ToImmutableArray();
+        }
+
+        private static async Task<ImmutableArray<UnitTestingDocumentSpan>> GetSourceLocationsInProcessAsync(
+            Document document,
+            string container,
+            string symbolName,
+            int symbolArity,
+            UnitTestingSearchQuery query,
+            CancellationToken cancellationToken)
+        {
+            var index = await TopLevelSyntaxTreeIndex.GetIndexAsync(document, cancellationToken).ConfigureAwait(false);
+            if (index == null)
+                return ImmutableArray<UnitTestingDocumentSpan>.Empty;
+
+            using var _ = ArrayBuilder<UnitTestingDocumentSpan>.GetInstance(out var result);
+
+            SyntaxTree? tree = null;
+
+            // Walk each of the top-level-index infos we've got for this tree.
+            foreach (var info in index.DeclaredSymbolInfos)
+            {
+                // Fast check to see if this looks like a candidate.
+                if (info.TypeParameterCount == symbolArity &&
+                    info.Name == symbolName)
+                {
+                    // If it's a method, the parameter count much match.
+                    if (query.MethodName != null)
+                    {
+                        if (info.Kind is not (DeclaredSymbolInfoKind.Method or DeclaredSymbolInfoKind.ExtensionMethod) ||
+                            info.ParameterCount != query.MethodParameterCount)
+                        {
+                            continue;
+                        }
+                    }
+
+                    // Looking promising so far.  Check that the container matches what the caller needs.
+                    if (info.FullyQualifiedContainerName != container)
+                        continue;
+
+                    // Unit testing needs to know the final span a location may be mapped to (e.g. with `#line` taken
+                    // into consideration).  So map that information here for them.
+                    tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
looks like this always will call `GetRequiredSyntaxTreeAsync` and `tree` doesnt look like its accessed outside the loop.  Could remove the definition on 139 and make this var?

Reply:
good catch.  this was meant to be `??=`. have fixed.